### ISSUE TITLE
[BUG] `planInformation()` of `ReservationOrderResponse` always returns `null`

### ISSUE THREAD

**Issue by Tomasz-Kluczkowski** [Azure/azure-sdk-for-java#37813]
**Describe the bug**

`planInformation()` method of `ReservationOrderResponse` always returns `null`.
All other reservation order data is present on all returned items from result of calling `manager.reservationOrders().list()` and iterating/viewing it in debug mode.

***Exception or Stack Trace***
Add the exception log and stack trace if available

N/A

**To Reproduce**

Run the code snippet, all other reservation order data is present but the billing plan information.

***Code Snippet***
Add the code snippet that causes the issue.

```java
public class GetCosts {
    public static void main(String[] args) {
        AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
        TokenCredential credential = new DefaultAzureCredentialBuilder()
                .authorityHost(profile.getEnvironment().getActiveDirectoryEndpoint())
                .build();
        ReservationsManager manager = ReservationsManager
                .authenticate(credential, profile);

        manager.reservationOrders().list().forEach(
                reservationOrder -> System.out.println(reservationOrder.planInformation())
                );
    }
}
```
**Expected behavior**

call to `planInformation` of `ReservationOrderResponse` contains plan information as described here: https://learn.microsoft.com/en-us/java/api/com.azure.resourcemanager.reservations.models.reservationorderbillingplaninformation?view=azure-java-preview

**Screenshots**

Output of the above snippet:

```bash
17:12:22.359 [main] INFO com.azure.identity.ChainedTokenCredential -- Azure Identity => Attempted credential EnvironmentCredential is unavailable.
17:12:22.361 [main] INFO com.azure.identity.ChainedTokenCredential -- Azure Identity => Attempted credential WorkloadIdentityCredential is unavailable.
17:12:22.924 [ForkJoinPool.commonPool-worker-1] INFO com.azure.identity.ChainedTokenCredential -- Azure Identity => Attempted credential ManagedIdentityCredential is unavailable.
17:12:22.930 [ForkJoinPool.commonPool-worker-1] INFO com.azure.identity.ChainedTokenCredential -- Azure Identity => Attempted credential SharedTokenCacheCredential is unavailable.
17:12:23.074 [ForkJoinPool.commonPool-worker-1] INFO com.azure.identity.ChainedTokenCredential -- Azure Identity => Attempted credential IntelliJCredential is unavailable.
17:12:24.063 [ForkJoinPool.commonPool-worker-1] INFO com.azure.identity.AzureCliCredential -- Azure Identity => getToken() result for scopes [https://management.core.windows.net//.default]: SUCCESS
17:12:24.063 [ForkJoinPool.commonPool-worker-1] INFO com.azure.identity.ChainedTokenCredential -- Azure Identity => Attempted credential AzureCliCredential returns a token
17:12:24.065 [ForkJoinPool.commonPool-worker-1] INFO com.azure.core.implementation.AccessTokenCache -- {"az.sdk.message":"Acquired a new access token."}
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
null
```

**Setup (please complete the following information):**
 - OS: [e.g. iOS] Mac OS Sonoma 14.1 (23B74)
 - IDE: [e.g. IntelliJ] IntelliJ
 - Library/Libraries: [e.g. com.azure:azure-core:1.16.0 (groupId:artifactId:version)] azure-resourcemanager-reservations, 1.0.0-beta.2
 - Java version: [e.g. 8] openjdk 21.0.1 2023-10-17
 - App Server/Environment: [e.g. Tomcat, WildFly, Azure Function, Apache Spark, Databricks, IDE plugin or anything special]
 - Frameworks: [e.g. Spring Boot, Micronaut, Quarkus, etc] Spring Boo

If you suspect a dependency version mismatch (e.g. you see `NoClassDefFoundError`, `NoSuchMethodError` or similar), please check out [Troubleshoot dependency version conflict article](https://aka.ms/azsdk/java/dependency/troubleshoot) first. If it doesn't provide solution for the problem, please provide:
- verbose dependency tree (`mvn dependency:tree -Dverbose`)
- exception message, full stack trace, and any available logs

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [ ] Bug Description Added
- [ ] Repro Steps Added
- [ ] Setup information Added

**Comment by joshfree** [comment]
Thanks for filing this github issue, @Tomasz-Kluczkowski.  @weidongxu-microsoft could you please follow up?

**Comment by weidongxu-microsoft** [comment]
Some service do include a simple model when getting from a `List` (maybe to save service time to fetch the details of each item).
It is indeed hard to tell from API.

Some service would also support an `expand` query param to let user optionally include details of the data. But this appears not in the "reservation" https://github.com/Azure/azure-rest-api-specs/blob/main/specification/reservations/resource-manager/Microsoft.Capacity/stable/2022-11-01/reservations.json#L235-L271

**Comment by Tomasz-Kluczkowski** [comment]
Yes, I appreciate that not all data will be populated, therefore I think best is to improve documentation and other issues with java sdk api I have found during this little experiment.

- the id attribute of reservation order (from a list call to get all reservation orders) cannot be used in get or getWithResponse methods to get reservation order details. It has to be split and only the uuid part used. It should be accepted by get/getWithResponse and the java sdk api should do whatever is necessary to make it work.

- the attributes that will not be populated in certain method calls should be described in those methods and a link to how to populate them mentioned

- special parameters like expand should have all possible values and their influence on the method described (one should not spend hours digging through REST api docs to figure it out, it should be documented in the java sdk).

- context should be optional as it is merged deeper in the code anyway (in getWithResponse for example)

Hopefully implementing this is possible and will make it easier for someone else 😀.

**Comment by weidongxu-microsoft** [comment]
For 1st bullet point, the general ARM would specify that the `name` in item in List can be used for the `get` call.
(the `id` should be something of `/providers/Microsoft.Capacity/reservationOrders/{name}`, with `name` as above)

For 4th bullet point, the `context` as required was a design decision that applied to all SDKs within the repo. You can supply a `Context.NONE` if nothing to input.

Agree to the rest 2 items. Since the SDK is generated from the Swagger, we will ask service to improve the documentation.

For the `expand`, the ideal case should be an Enum (or something like it) of List of Enum. But give that service had "string" there for the stable api-version, we likely only able to ask them to give more information in API description. 
Logged issue to service https://github.com/Azure/azure-rest-api-specs/issues/26860

**Comment by samvaity** [comment]
@weidongxu-microsoft / @XiaofeiCao / @v-hongli1  Could you please confirm if this is still an issue?

**Comment by weidongxu-microsoft** [comment]
We can follow up the service issue on specs issue https://github.com/Azure/azure-rest-api-specs/issues/26860
